util/helm/client.go

```diff
@@ -192,34 +198,34 @@ func (c *nativeHelmChart) ExtractChart(chart string, version string, project str
 			// 'helm pull' ensures that chart is downloaded into temp directory
 			_, err = helmCmd.PullOCI(c.repoURL, chart, version, tempDest, c.creds)
 			if err != nil {
-				return "", nil, err
+				return "", nil, fmt.Errorf("error pulling OCI chart: %w", err)
 			}
 		} else {
 			_, err = helmCmd.Fetch(c.repoURL, chart, version, tempDest, c.creds, passCredentials)
 			if err != nil {
-				return "", nil, err
+				return "", nil, fmt.Errorf("error fetching chart: %w", err)
 			}
 		}
 
 		// 'helm pull/fetch' file downloads chart into the tgz file and we move that to where we want it
 		infos, err := os.ReadDir(tempDest)
 		if err != nil {
-			return "", nil, err
+			return "", nil, fmt.Errorf("error reading directory %s: %w", tempDest, err)
 		}
 		if len(infos) != 1 {
 			return "", nil, fmt.Errorf("expected 1 file, found %v", len(infos))
 		}
 
 		err = os.Rename(filepath.Join(tempDest, infos[0].Name()), cachedChartPath)
 		if err != nil {
-			return "", nil, err
+			return "", nil, fmt.Errorf("error renaming file from %s to %s: %w", filepath.Join(tempDest, infos[0].Name()), cachedChartPath, err)
```

Comment:
Should we save `filepath.Join(tempDest, infos[0].Name())` into a variable and use it on line 219 as well?

Reply:
Thank you so much. I think that's a good idea 😃